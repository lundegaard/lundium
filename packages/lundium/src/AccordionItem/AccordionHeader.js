import React from 'react';
import PropTypes from 'prop-types';
import { cx, noop } from 'ramda-extension';

const AccordionHeader = ({
	index,
	indexOpen,
	setIndexOpen = noop,
	dark,
	highlighted,
	...props
}) => (
	<button
		className={cx(
			'accordion-header',
			{ 'accordion-header-dark': dark },
			{ 'accordion-header--open': index === indexOpen },
			{ highlighted }
		)}
		onClick={() => setIndexOpen(index === indexOpen ? undefined : index)}
		{...props}
	/>
);

AccordionHeader.propTypes = {
	dark: PropTypes.bool,
	highlighted: PropTypes.bool,
	index: PropTypes.number,
	indexOpen: PropTypes.number,
	setIndexOpen: PropTypes.func,
};

export default AccordionHeader;
