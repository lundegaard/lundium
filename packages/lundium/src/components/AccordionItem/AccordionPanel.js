import React from 'react';
import PropTypes from 'prop-types';
import { cx } from 'ramda-extension';

import Box from '../Box';

const AccordionPanel = ({ index = 0, indexOpen, ...props }) => (
	<Box
		className={cx('accordion-panel', { 'accordion-panel--open': index === indexOpen })}
		{...props}
	/>
);

AccordionPanel.propTypes = {
	index: PropTypes.number,
	indexOpen: PropTypes.number,
};

export default AccordionPanel;
