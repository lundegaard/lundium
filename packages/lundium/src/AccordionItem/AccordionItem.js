import React from 'react';
import PropTypes from 'prop-types';
import { noop } from 'ramda-extension';

import Box from '../Box';
import Icon from '../Icon';

import AccordionHeader from './AccordionHeader';
import AccordionPanel from './AccordionPanel';

const AccordionItem = ({ header, panel, dark, setIndexOpen = noop, highlighted, ...props }) => (
	<Box className="accordion-item">
		<AccordionHeader dark={dark} setIndexOpen={setIndexOpen} highlighted={highlighted} {...props}>
			{header}
			<Icon type="caret-down" size={6} />
		</AccordionHeader>
		<AccordionPanel {...props}>{panel}</AccordionPanel>
	</Box>
);

AccordionItem.propTypes = {
	/** If set, dark mode is enabled. */
	dark: PropTypes.bool,
	/** Header to be rendered. */
	header: PropTypes.node,
	/** If set, item is highlighted. */
	highlighted: PropTypes.bool,
	/** Panel to be rendered. */
	panel: PropTypes.node,
	/** Function for setting index to open. */
	setIndexOpen: PropTypes.func,
};

export default AccordionItem;
