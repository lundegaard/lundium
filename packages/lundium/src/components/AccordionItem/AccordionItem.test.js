import React from 'react';
import { shallow } from 'enzyme';

import AccordionItem from './AccordionItem';

describe('<AccordionItem />', () => {
	const requiredProps = {};

	it('matches snapshot when shallowly rendered', () => {
		const wrapper = shallow(<AccordionItem {...requiredProps} />);
		expect(wrapper).toMatchSnapshot();
	});
});
