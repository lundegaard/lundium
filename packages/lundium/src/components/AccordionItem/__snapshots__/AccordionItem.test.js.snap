// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccordionItem /> matches snapshot when shallowly rendered 1`] = `
<forwardRef(Box)
  className="accordion-item"
>
  <AccordionHeader
    setIndexOpen={[Function]}
  >
    <Icon
      size={6}
      type="caret-down"
    />
  </AccordionHeader>
  <AccordionPanel />
</forwardRef(Box)>
`;
